addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/predefined/HashProcessor.java

```diff
@@ -32,7 +35,13 @@ public HashProcessor(String algorithm) {
 
     @Override
     protected String processInternal(String value) throws NoSuchAlgorithmException {
-        return HexStringEncoder.getHexString(getHash(value.getBytes()));
+        EncodeDecodeOptions encDecOpts =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionEncoder.class)
+                        .getOptions();
+        String output = HexStringEncoder.getHexString(getHash(value.getBytes()));
```

Comment:
IMO we should stop accessing the options directly and chain the processors (we can have a processor that lowers/uppers case the result), what's chosen in the UI should not force everywhere to be one way or the other (since we are now exposing this to the scripts it should be up to them to lower case or upper case as needed).

In any case it can be something to other PR.

Reply:
I do actually have another branch with lowercase, uppercase, and reverse processors.

The option for implemented here does only impact hashers. (To address the ticket and poll.)

I have been mulling over the idea of adding a tab that allows users to create a chain or recipe to sequentially apply processors.